bin/ruby-build

```diff
@@ -1022,20 +1023,28 @@ locate_llvm() {
 }
 
 needs_yaml() {
-  [[ "$RUBY_CONFIGURE_OPTS" != *--with-libyaml-dir=* ]] &&
-  ! use_homebrew_yaml
+  [ -z "$(homebrew_yaml_dir)" ]
```

Comment:
For backwards compatibility, please restore to the old implementation of `needs_yaml`. There was no need to change it, and the new implementation doesn't print the line `ruby-build: using libyaml from homebrew` anymore.